test/openshift/e2e/operator_expectations.go

```diff
@@ -235,48 +255,28 @@ func (tc *testConfig) ExpectAutoscalerScalesOut() error {
 		glog.Info("Expect at least a new node to come up")
 		glog.Infof("Initial number of nodes: %d. New number of nodes: %d", clusterInitialTotalNodes, len(nodeList.Items))
 		return len(nodeList.Items) > clusterInitialTotalNodes, nil
-	})
+	}); err != nil {
+		return err
+	}
 
 	glog.Info("Delete workload")
-	err = wait.PollImmediate(1*time.Second, waitMedium, func() (bool, error) {
+	if err := wait.PollImmediate(1*time.Second, waitMedium, func() (bool, error) {
 		if err := tc.client.Delete(context.TODO(), &workLoad); err != nil {
 			glog.Errorf("error querying api for workLoad object: %v, retrying...", err)
 			return false, nil
 		}
 		return true, nil
-	})
-	if err != nil {
+	}); err != nil {
 		return err
 	}
 
-	// We delete the clusterAutoscaler and ensure the initial number of replicas to get the cluster to the initial number of nodes
-	// TODO: validate the autoscaler to scale down
-	glog.Info("Delete clusterAutoscaler object")
-	err = wait.PollImmediate(1*time.Second, waitShort, func() (bool, error) {
-		if err := tc.client.Delete(context.TODO(), &clusterAutoscaler); err != nil {
-			glog.Errorf("error querying api for clusterAutoscaler object: %v, retrying...", err)
-			return false, nil
-		}
-		return true, nil
-	})
-	if err != nil {
-		return err
-	}
-
-	glog.Info("Delete machineAutoscaler object")
-	err = wait.PollImmediate(1*time.Second, waitShort, func() (bool, error) {
-		if err := tc.client.Delete(context.TODO(), &machineAutoscaler); err != nil {
-			glog.Errorf("error querying api for machineAutoscaler object: %v, retrying...", err)
-			return false, nil
-		}
-		return true, nil
-	})
-	if err != nil {
-		return err
-	}
+	// As we have just deleted the workload the autoscaler will
```

Comment:
I'm not sure if deleting the workload (job) will remove the pods as well, if not the autoscaler will still try to allocate new resources instead of scaling down

Reply:
I believe it does. That's how I've done 100% of my testing over the last few months.